### PR TITLE
add KubeArmorController

### LINKED ISSUE #241: Merge kubearmor-host-policy-manager into kubearmor-policy-manager
we're currently using two managers for KubeArmorPolicy and KubeArmorHostPolicy.
It would be better if we have a single manager for both policies.

### PR DESCRIPTION
**Pull request by achrefbensaad**
Signed-off-by: achref ben saad <achref9612@gmail.com>

### CODE REVIEW

File: deployments/controller/kubearmor-controller-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,1261 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.1
+  creationTimestamp: null
+  name: kubearmorhostpolicies.security.kubearmor.com
```
Comment (nam-jaehyun) [inline deployments/controller/kubearmor-controller-manager.yaml:7]: kubearmorhostpolicies?

File: deployments/main.go [hunk 2]
```diff
@@ -34,12 +33,8 @@ func main() {
 			dp.GetRelayService(namespace),
 			dp.GetRelayDeployment(namespace),
 			dp.GenerateDaemonSet(strings.ToLower(env), namespace),
-			dp.GetPolicyManagerService(namespace),
```
Comment (nam-jaehyun) [inline deployments/main.go:37]: no controller? only CRDs?

File: pkg/KubeArmorController/api/v1/kubearmorhostpolicy_types.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KubeArmorHostPolicySpec defines the desired state of KubeArmorHostPolicy
+type KubeArmorHostPolicySpec struct {
+	NodeSelector NodeSelectorType `json:"nodeSelector"`
+
+	Process      ProcessType          `json:"process,omitempty"`
```
Comment (nam-jaehyun) [inline pkg/KubeArmorController/api/v1/kubearmorhostpolicy_types.go:14]: hsp doesn't support the global allow policy, which means that hsp with allow action must have fromSource.
this is the big difference between hsp and ksp. did you consider this point?

**Review by nam-jaehyun — CHANGES_REQUESTED** [review]
Please check if there are wrong names that you missed to change while merging three controllers together. Also, please check if you correctly bring the functions and intents for each controller to this unified controller. I can still see some incorrect points.

File: deployments/controller/kubearmor-controller-manager.yaml (hunk 1 above)
Comment (achrefbensaad) [reply deployments/controller/kubearmor-controller-manager.yaml:7]: thats the plural name of the CRD.

File: deployments/main.go (hunk 2 above)
Comment (achrefbensaad) [reply deployments/main.go:37]: Yes as the controller will require certificates pair to operate and we cannot store  those on git.

However we can update the helm chart to generate them at the during the install process.

File: pkg/KubeArmorController/api/v1/kubearmorhostpolicy_types.go (hunk 3 above)
Comment (achrefbensaad) [reply pkg/KubeArmorController/api/v1/kubearmorhostpolicy_types.go:14]: I am not aware of this detail. However, the original host policy controller have the same structure. 
https://github.com/kubearmor/KubeArmor/blob/main/pkg/KubeArmorHostPolicy/api/security.kubearmor.com/v1/kubearmorhostpolicy_types.go#L240-L258

**Comment by achrefbensaad** [comment]
> Please check if there are wrong names that you missed to change while merging three controllers together. Also, please check if you correctly bring the functions and intents for each controller to this unified controller. I can still see some incorrect points.

Can you please point me to the incorrect points, I ran the `make manifests` and the results are a carbone copy of the original.

**Review by nyrahul — APPROVED** [review]
LGTM

**Comment by nyrahul** [comment]
Hey @nam-jaehyun , can you please revisit your comments and @achrefbensaad responses ... and provide a signal to go ahead/no-go with this?

**Pull request merged**